src/core/server/saved_objects/migrations/core/index_migrator.ts

```diff
@@ -45,23 +46,66 @@ export class IndexMigrator {
    * @returns {Promise<MigrationResult>}
    */
   public async migrate(): Promise<MigrationResult> {
-    const context = await migrationContext(this.opts);
+    const { dryRun, index, callCluster, log, batchSize } = this.opts;
+    if (dryRun) {
+      const indexInfo = await fetchInfo(callCluster, index);
+      if (indexInfo.exists === false) {
+        return {
+          status: 'skipped',
+          alias: this.opts.index,
+          reason: `nothing to migrate, index ${index} doesn't exist.`,
+        };
+      } else if (!indexInfo.aliases[index]) {
+        return {
+          status: 'skipped',
+          alias: index,
+          reason: `expected an alias but found an index: ${index}.`,
+        };
+      } else if (await requiresMigration(await migrationContext(this.opts))) {
+        const dryRunAlias = index + '_dry_run';
+        const dryRunIndex = dryRunAlias + '_1';
+        log.info(
+          `Reindexing ${indexInfo.indexName} into ${dryRunIndex} for performing dry run migration.`
+        );
+        await Index.createIndex(callCluster, dryRunIndex, indexInfo.mappings, dryRunAlias);
+        await Index.reindex(callCluster, indexInfo.indexName, dryRunIndex, batchSize);
+      } else {
+        return { status: 'skipped', alias: index };
+      }
+    }
 
-    return coordinateMigration({
-      log: context.log,
+    const context = await migrationContext({
+      ...this.opts,
+      index: this.opts.index + (dryRun ? '_dry_run' : ''),
+    });
```

Comment:
Do we have a mechanism to remove all the `_dry_run` indexes at startup? Thinking about the edge case where the dry run migration fails (for any reason) and therefor do not delete the associated index before process termination.

Reply:
I'm scared of accidentally deleting an index that wasn't part of a dry run. So by only doing it after a dry run migration, we're sure that we created the index we're deleting.

This is annoying, but we will display an error message to tell the user to remove those indexes.